distributed/scheduler.py

```diff
@@ -2012,6 +2014,31 @@ def update_graph(
         if isinstance(user_priority, Number):
             user_priority = {k: user_priority for k in tasks}
 
+        annotations = annotations or {}
+        restrictions = restrictions or {}
+        loose_restrictions = loose_restrictions or []
+        resources = resources or {}
+        retries = retries or {}
+
+        # Override existing taxonomy with per task annotations
+        # https://stackoverflow.com/a/20308657/1611416
+        if annotations:
+            for k, a in annotations.items():
+                if "priority" in a:
+                    priority[k] = a["priority"]
+
+                if "workers" in a:
+                    restrictions[k] = a["workers"]
+
+                if a.get("allow_other_workers", False):
+                    loose_restrictions.append(k)
+
+                if "retries" in a:
+                    retries[k] = a["retries"]
+
+                if "resources" in a:
+                    resources[k] = a["resources"]
```

Comment:
See comment in https://github.com/dask/dask/pull/6889/files#r534262218

Reply:
Updated with tranposed annotations! Thanks for the suggestion.